### ISSUE TITLE
Use exemplar for Timer max value.

### ISSUE THREAD

**Issue by jtama** [micrometer-metrics/micrometer#3485]
**Please describe the feature request.**
If I understand correctly, exemplars are only collected for timer with distributionStatistics set to "prometheus". I think it wille be usefull to have exemplar for max value in a timer even if histogram are not enabled.

**Rationale**
I wan't to be able to easily correlate the longest occurences with my traces and logs.

**Comment by jonatan-ivanov** [comment]
**TL;DR:** You can simulate this with a histogram and calculating the 100%'ile (using `histogram_quantile `).

**Details:**
Right now we support Exemplars as defined in the [OpenMetrics specification](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md). This means Prometheus counters and histogram buckets.

Timer's max value is a Prometheus gauge, that's the reason why Exemplars are missing right now from `max`: OpenMetrics does not support Exemplars for gauges.

Other than this, I'm with you, I don't have the full context for OpenMetrics specs but I find having Exemplars on `max` useful.

There is a PR and a separate discussion around a similar functionality (lack of Exemplars on `count` for Timer/Distribution): https://github.com/micrometer-metrics/micrometer/pull/3460
Let me include this in the discussion there to see if we can put exemplars on gauges.

**Comment by jonatan-ivanov** [comment]
Blocked on the outcome of https://github.com/micrometer-metrics/micrometer/pull/3460
